packages/libs/web-common/src/components/records/ClustalAlignmentForm.tsx

```diff
@@ -0,0 +1,171 @@
+import { Dialog } from '@veupathdb/wdk-client/lib/Components';
+
+interface ClustalAlignmentFormProps {
+  action: string;
+  sequenceCount: number;
+  children: React.ReactNode;
+  sequenceType?: string;
+  warnThreshold?: number | ((form: HTMLFormElement) => number);
+  blockThreshold?: number | ((form: HTMLFormElement) => number);
+}
+
+const DEFAULT_WARN_THRESHOLD = 50;
+const DEFAULT_BLOCK_THRESHOLD = 1000;
+
+export default function ClustalAlignmentForm({
+  action,
+  sequenceCount,
+  children,
+  sequenceType = 'sequences',
+  warnThreshold,
+  blockThreshold,
+}: ClustalAlignmentFormProps) {
+  const [showModal, setShowModal] = useState(false);
+  const [evaluatedWarnThreshold, setEvaluatedWarnThreshold] =
+    useState<number | null>(null);
+  const [evaluatedBlockThreshold, setEvaluatedBlockThreshold] =
+    useState<number | null>(null);
+  const formRef = useRef<HTMLFormElement>(null);
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    // Evaluate thresholds dynamically if they're functions
+    const actualWarn =
+      typeof warnThreshold === 'function'
+        ? formRef.current
+          ? warnThreshold(formRef.current)
+          : DEFAULT_WARN_THRESHOLD
+        : warnThreshold ?? DEFAULT_WARN_THRESHOLD;
+    const actualBlock =
+      typeof blockThreshold === 'function'
+        ? formRef.current
+          ? blockThreshold(formRef.current)
+          : DEFAULT_BLOCK_THRESHOLD
+        : blockThreshold ?? DEFAULT_BLOCK_THRESHOLD;
+
+    setEvaluatedWarnThreshold(actualWarn);
+    setEvaluatedBlockThreshold(actualBlock);
+    setShowModal(true);
+  };
+
+  const handleConfirm = () => {
+    setShowModal(false);
+    if (formRef.current) {
+      formRef.current.submit();
+    }
+  };
+
+  const handleCancel = () => {
+    setShowModal(false);
+  };
+
+  const isBlocked =
+    evaluatedBlockThreshold !== null && sequenceCount > evaluatedBlockThreshold;
+  const showWarning =
+    evaluatedWarnThreshold !== null &&
+    evaluatedBlockThreshold !== null &&
+    sequenceCount > evaluatedWarnThreshold &&
+    sequenceCount <= evaluatedBlockThreshold;
+
+  return (
+    <>
+      <form
+        ref={formRef}
+        action={action}
+        target="_blank"
+        method="post"
+        onSubmit={handleSubmit}
+      >
+        {children}
+      </form>
+
+      <Dialog
+        open={showModal}
+        modal
+        title="Run Clustal Omega Alignment"
+        onClose={handleCancel}
+      >
+        <div style={{ padding: '10px', maxWidth: '600px' }}>
+          <p style={{ marginTop: '0px' }}>
+            The alignment results will open in a new browser tab.
+          </p>
+          <p>
+            The tab may appear empty while the alignment is running. Please be
+            patient and avoid resubmitting - multiple requests will not make it
+            faster and can overload our servers.
+          </p>
+
+          {showWarning && (
+            <p
```

Comment:
Idk if this is out of scope, but it seems like the warning should come at the top? So the warning comes first in the dialog and then the regular text follows